src/masternodes/rpc_icxorderbook.cpp

```diff
@@ -0,0 +1,1442 @@
+                },
+                RPCResult{
+                        "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
+                },
+                RPCExamples{
+                        HelpExampleCli("icx_submitdfchtlc", "'{\"offerTx\":\"<txid>\",\"amount\":\"10\","
+                                                        "\"receiveAddress\":\"<address>\",\"hash\":\"<hash>\",\"timeout\":\"50\"}'")
+                },
+     }.Check(request);
+
+    if (pwallet->chain().isInitialBlockDownload())
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, "Cannot submit dfc htlc while still in Initial Block Download");
+
+    pwallet->BlockUntilSyncedToCurrentChain();
+    LockedCoinsScopedGuard lcGuard(pwallet);
+
+    RPCTypeCheck(request.params, {UniValue::VOBJ}, false);
+    if (request.params[0].isNull())
+        throw JSONRPCError(RPC_INVALID_PARAMETER,
+                           "Invalid parameters, arguments 1 must be non-null and expected as object at least with "
+                           "{\"offerTx\",\"amount\",\"receiverAddress\",\"hash\"}");
+
+    UniValue metaObj = request.params[0].get_obj();
+    UniValue const & txInputs = request.params[1];
+
+    CICXSubmitDFCHTLC submitdfchtlc;
+
+    if (!metaObj["offerTx"].isNull())
+        submitdfchtlc.offerTx = uint256S(metaObj["offerTx"].getValStr());
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"offerTx\" must be non-null");
+    if (!metaObj["amount"].isNull())
+        submitdfchtlc.amount = AmountFromValue(metaObj["amount"]);
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"amount\" must be non-null");
+    if (!metaObj["receiveAddress"].isNull())
+        submitdfchtlc.receiveAddress = DecodeScript(metaObj["receiveAddress"].getValStr());
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"receiveAddress\" must be non-null");
+    if (!metaObj["hash"].isNull())
+        submitdfchtlc.hash = uint256S(metaObj["hash"].getValStr());
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"hash\" must be non-null");
+    if (!metaObj["timeout"].isNull())
+        submitdfchtlc.timeout = metaObj["timeout"].get_int();
+
+    int targetHeight;
+    {
+        LOCK(cs_main);
+        auto offer = pcustomcsview->GetICXMakeOfferByCreationTx(submitdfchtlc.offerTx);
+        if (!offer)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("offerTx (%s) does not exist",submitdfchtlc.offerTx.GetHex()));
+        auto order = pcustomcsview->GetICXOrderByCreationTx(offer->orderTx);
+        if (!order)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("orderTx (%s) does not exist",offer->orderTx.GetHex()));
+
+        if (order->orderType == CICXOrder::TYPE_INTERNAL)
+        {
+            if (submitdfchtlc.amount * order->orderPrice / COIN != offer->amount)
```

Comment:
`submitdfchtlc.amount * order->orderPrice` may overflow, need to convert to 256bits integer to calculate.